### PR TITLE
JMX: Allow specifying multiple target system scripts

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
**Description:**
~~Requires https://github.com/open-telemetry/opentelemetry-java-contrib/pull/27~~

Feature addition - These changes add comma-separated value support for `otel.jmx.target.system` so that multiple builtin scripts can be run against the same mbean server.  This is useful when wanting* general JVM metrics along with a more specific target system.

**Existing Issue(s):**

Resolves https://github.com/open-telemetry/opentelemetry-java-contrib/issues/26

**Testing:**

Updates existing unit tests and adds integration suite.

**Documentation:**

Updates existing documentation

**Outstanding items:**

Doesn't add default target system of `jvm`, which will be introduced in a later change to resolve https://github.com/open-telemetry/opentelemetry-java-contrib/issues/12

### CODE REVIEW

File: contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java [hunk 1]
```diff
@@ -69,6 +71,10 @@
     groovyScript = properties.getProperty(GROOVY_SCRIPT);
     targetSystem = properties.getProperty(TARGET_SYSTEM, "").toLowerCase().trim();
 
+    List<String> targets =
+        Arrays.asList(isBlank(targetSystem) ? new String[0] : targetSystem.split(","));
+    targetSystems = new LinkedHashSet<>(targets).toArray(new String[0]);
```
Comment (pmcollins) [inline contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:76]: Is this to remove duplicates?

Comment (rmfitzpatrick) [reply contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:76]: Right, it's my understanding that a `LinkedHashSet` preserves insertion order as well.

File: contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java [hunk 2]
```diff
@@ -117,9 +123,12 @@ void validate() {
           "Only one of " + GROOVY_SCRIPT + " or " + TARGET_SYSTEM + " can be specified.");
     }
 
-    if (!isBlank(targetSystem) && !AVAILABLE_TARGET_SYSTEMS.contains(targetSystem)) {
-      throw new ConfigurationException(
-          String.format("%s must be one of %s", targetSystem, AVAILABLE_TARGET_SYSTEMS));
+    for (final String target : targetSystems) {
+      if (!AVAILABLE_TARGET_SYSTEMS.contains(target)) {
```
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java]: It's a List, so changing `targetSystems` as suggested works.  updated

File: contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java]: Indeed, updated.

**Review by pmcollins — APPROVED** [review]
LGTM

**Pull request merged**